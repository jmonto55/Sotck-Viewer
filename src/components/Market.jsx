import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { v4 as uuidv4 } from 'uuid';
import { Link } from 'react-router-dom';
import { fetchStocks, filterStocks, selectStock } from '../redux/markets/marketsSlice';
import stock from '../assets/stock.svg';

const Market = () => {
  const dispatch = useDispatch();
  const { filteredList, isLoading, market } = useSelector((store) => store.markets);

  const handleScrollToTop = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  useEffect(() => {
    dispatch(fetchStocks(market));
    window.scrollTo(0, 0);
  }, [dispatch, market]);

  if (isLoading) {
    return (
      <div className="opacity-50 w-screen h-screen flex flex-col justify-center items-center">
        <div className="animate-spin border-8 border-t-transparent mb-10 border-white w-24 h-24 rounded-full" />
        <h1 className="text-xl text-white animate-ping">Loading...</h1>
      </div>
    );
  }
  return (
    <div className="w-screen h-full flex max-w-screen-2xl flex-col my-20 items-center">
      <button onClick={handleScrollToTop} type="button" className="animate-bounce-slow hover:opacity-50 shadow-[0_4px_8px_rgba(0,0,0,0.5)] bg-neutral-800 z-20 text-white flex w-12 h-12 fixed bottom-3 right-3 rounded-full">
        <span className="m-auto material-symbols-outlined">
          expand_less
        </span>
      </button>
      <header className="m-auto mt-3 max-w-screen-2xl bg-white w-full bg-opacity-10 h-48 p-2 md:py-10 md:pt-14 lg:pr-24 rounded-t-lg flex justify-between items-center relative">
        <Link to="/" className="text-white material-symbols-outlined text-xl cursor-pointer absolute top-4 left-4 md:top-6 md:left-6 opacity-50 hover:scale-110">
          arrow_back_ios
        </Link>
        <h2 className={`animate-pulse-slow text-center w-1/2 ${market === 'EURONEXT' || market === 'NASDAQ' ? 'text-4xl' : 'text-6xl'} md:text-6xl text-white`}>{market}</h2>
        <img src={stock} alt="stock" className="md:mr-20 animate-pulse-slow mt-10 w-40 md:w-64" />
      </header>
      <input
        type="text"
        placeholder="Filter by company name or symbol"
        className="z-20 max-w-screen-2xl pt-4 w-full placeholder-shown:bg-neutral-500 p-2 focus:outline-none focus:bg-neutral-500 focus:text-white"
        onChange={(e) => dispatch(filterStocks(e.target.value))}
      />
      <ul className="z-10 w-full grid grid-cols-2 mt-6">
        {filteredList.map((stock) => (
          <Link
            to="/stock"
            className="h-32 shadow-[0_4px_8px_rgba(0,0,0,0.5)] cursor-pointer hover:scale-y-105 text-center text-white p-3 flex flex-col justify-between"
            key={uuidv4()}
            onClick={() => dispatch(selectStock(stock))}
          >
            <span className="material-symbols-outlined text-base absolute top-4 right-4 opacity-50 hover:scale-110">
              arrow_forward_ios
            </span>
            <h2 className="text-4xl">{stock.symbol}</h2>
            <p className="text-left">
              <span>$</span>
              {stock.price || 125.0}
            </p>
            <h2 className="text-left text-sm">{stock.companyName}</h2>
          </Link>
        ))}
      </ul>
    </div>
  );
};

export default Market;
