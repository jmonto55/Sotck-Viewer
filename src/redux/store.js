import { configureStore } from '@reduxjs/toolkit';
import marketsReducer from './markets/marketsSlice';

const store = configureStore({
  reducer: {
    markets: marketsReducer,
  },
});

export default store;
