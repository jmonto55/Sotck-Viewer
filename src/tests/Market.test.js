import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Market from './__mocks__/Market';
import store from '../redux/store';
import '@testing-library/jest-dom';

describe('Test suite for Market component', () => {
  it('Should render a Snapshot of Market component', () => {
    const market = render(
      <Provider store={store}>
        <MemoryRouter>
          <Market />
        </MemoryRouter>
      </Provider>,
    );
    expect(market).toMatchSnapshot();
  });
  it('Should test if Markets section is being rendered', () => {
    render(
      <Provider store={store}>
        <MemoryRouter>
          <Market />
        </MemoryRouter>
      </Provider>,
    );

    expect(screen.getByText('expand_less')).toBeInTheDocument();
  });
});
