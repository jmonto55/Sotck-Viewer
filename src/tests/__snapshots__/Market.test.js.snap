// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for Market component Should render a Snapshot of Market component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-full flex max-w-screen-2xl flex-col my-20 items-center"
      >
        <button
          class="animate-bounce-slow hover:opacity-50 shadow-[0_4px_8px_rgba(0,0,0,0.5)] bg-neutral-800 z-20 text-white flex w-12 h-12 fixed bottom-3 right-3 rounded-full"
          type="button"
        >
          <span
            class="m-auto material-symbols-outlined"
          >
            expand_less
          </span>
        </button>
        <header
          class="m-auto mt-3 max-w-screen-2xl bg-white w-full bg-opacity-10 h-48 p-2 md:py-10 md:pt-14 lg:pr-24 rounded-t-lg flex justify-between items-center relative"
        >
          <a
            class="text-white material-symbols-outlined text-xl cursor-pointer absolute top-4 left-4 md:top-6 md:left-6 opacity-50 hover:scale-110"
            href="/"
          >
            arrow_back_ios
          </a>
          <h2
            class="animate-pulse-slow text-center w-1/2 text-6xl md:text-6xl text-white"
          />
          <img
            alt="stock"
            class="md:mr-20 animate-pulse-slow mt-10 w-40 md:w-64"
            src="stock.svg"
          />
        </header>
        <input
          class="z-20 max-w-screen-2xl pt-4 w-full placeholder-shown:bg-neutral-500 p-2 focus:outline-none focus:bg-neutral-500 focus:text-white"
          placeholder="Filter by company name or symbol"
          type="text"
        />
        <ul
          class="z-10 w-full grid grid-cols-2 mt-6"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-full flex max-w-screen-2xl flex-col my-20 items-center"
    >
      <button
        class="animate-bounce-slow hover:opacity-50 shadow-[0_4px_8px_rgba(0,0,0,0.5)] bg-neutral-800 z-20 text-white flex w-12 h-12 fixed bottom-3 right-3 rounded-full"
        type="button"
      >
        <span
          class="m-auto material-symbols-outlined"
        >
          expand_less
        </span>
      </button>
      <header
        class="m-auto mt-3 max-w-screen-2xl bg-white w-full bg-opacity-10 h-48 p-2 md:py-10 md:pt-14 lg:pr-24 rounded-t-lg flex justify-between items-center relative"
      >
        <a
          class="text-white material-symbols-outlined text-xl cursor-pointer absolute top-4 left-4 md:top-6 md:left-6 opacity-50 hover:scale-110"
          href="/"
        >
          arrow_back_ios
        </a>
        <h2
          class="animate-pulse-slow text-center w-1/2 text-6xl md:text-6xl text-white"
        />
        <img
          alt="stock"
          class="md:mr-20 animate-pulse-slow mt-10 w-40 md:w-64"
          src="stock.svg"
        />
      </header>
      <input
        class="z-20 max-w-screen-2xl pt-4 w-full placeholder-shown:bg-neutral-500 p-2 focus:outline-none focus:bg-neutral-500 focus:text-white"
        placeholder="Filter by company name or symbol"
        type="text"
      />
      <ul
        class="z-10 w-full grid grid-cols-2 mt-6"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
